src/java/org/apache/cassandra/streaming/StreamTransferTask.java

```diff
@@ -62,6 +61,7 @@ public class StreamTransferTask extends StreamTask
     public StreamTransferTask(StreamSession session, TableId tableId)
     {
         super(session, tableId);
+        timeoutExecutor.setExecuteExistingDelayedTasksAfterShutdownPolicy(false);
```

Comment:
@dcapwell My thinking here is that given no non-test codepath actually shuts `timeoutExecutor` down, this should be a safe change.

Reply:
Also, when this timeout is scheduled in prod code...

```
task.scheduleTimeout(message.header.sequenceNumber, 12, TimeUnit.HOURS);
```

:D